plugin-filters/finaid-post-list-layout.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Provides post list layouts for media
+ */
+namespace FinAid\Utils\Plugins;
+
+/**
+ * Function for displaying the beginning of a media post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_media_before( $content, $posts, $atts ) {
+	ob_start();
+?>
+	<div class="ucf-post-list ucf-post-list-media" id="post-list-<?php echo $atts['list_id']; ?>">
+<?php
+	return ob_get_clean();
+}
+
+// add_filter( 'ucf_post_list_display_media_before', array( __NAMESPACE__ . "\ucf_post_list_display_media_before" ), 10, 3 );
+
+/**
+ * Function for displaying the title of the media post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_media_title( $content, $posts, $atts ) {
+	$formatted_title = '';
+
+	if ( $list_title = $atts['list_title'] ) {
+		$formatted_title = "<h2 class=\"ucf-post-list-title\">$list_title</h2>";
+	}
+
+	return $formatted_title;
+}
+
+add_filter( 'ucf_post_list_display_media_title', __NAMESPACE__ . '\ucf_post_list_display_media_title', 10, 3 );
+
+/**
+ * Function for displaying the items of the media post list
+ * @author Jim Barnes
+ * @since 1.0.0
+ * @param string $content The content passed to the filter
+ * @param array $posts The posts
+ * @param array $atts The shortcode arguments
+ * @return string
+ */
+function ucf_post_list_display_media( $content, $posts, $atts ) {
```

Comment:
Need to add a check at the top of this function that ensures `$posts` is an array, e.g.
`if ( ! is_array( $posts ) && $posts !== false ) { $posts = array( $posts ); }`

Logic in this function should also check to ensure `$posts` is not false-y before attempting to loop through it

Reply:
Added, along with a count check before the grouping.